### PR TITLE
mark tags

### PR DESCRIPTION
**Pull request by GJJ121785021**

### CODE REVIEW

**Review by waitan2018 — APPROVED** [review]

File: django_simple_api/utils.py [hunk 1]
```diff
@@ -49,3 +51,35 @@ def is_class_view(handler: Callable) -> bool:
...
+):
+    for item in urlpatterns:
+        if isinstance(item, URLPattern):
+            _wrapper_handler(wrappers, item.callback)
+        else:
+            wrapper_urlpatterns(wrappers, item.url_patterns)
+
+
+def wrapper_chain(wrappers: Sequence[Callable[[T], T]], view: T) -> T:
+    if isinstance(view, (list, tuple)):
```
Comment (abersheeran) [inline django_simple_api/utils.py:74]: `include` always return tuple.

File: django_simple_api/utils.py [hunk 2]
```diff
@@ -49,3 +51,35 @@ def is_class_view(handler: Callable) -> bool:
...
+        # For include(...) processing.
+        urlconf_module = view[0]
+        urlpatterns = getattr(urlconf_module, "urlpatterns", urlconf_module)
+        wrapper_urlpatterns(wrappers, urlpatterns)
+    elif callable(view):
+        view = _wrapper_handler(wrappers, view)
+    else:
+        raise TypeError(
+            "view must be a callable or a list/tuple in the case of include()."
```
Comment (abersheeran) [inline django_simple_api/utils.py:83]: a callable or a tuple in the case of `include()`

**Review by abersheeran — CHANGES_REQUESTED** [review]

File: django_simple_api/utils.py [hunk 3]
```diff
@@ -49,3 +51,35 @@ def is_class_view(handler: Callable) -> bool:
...
+    urlpatterns: List[Union[URLPattern, URLResolver]],
+):
+    for item in urlpatterns:
+        if isinstance(item, URLPattern):
+            _wrapper_handler(wrappers, item.callback)
+        else:
+            wrapper_urlpatterns(wrappers, item.url_patterns)
+
+
+def wrapper_chain(wrappers: Sequence[Callable[[T], T]], view: T) -> T:
```
Comment (abersheeran) [inline django_simple_api/utils.py:73]: `view: Any`

**Review by abersheeran — CHANGES_REQUESTED** [review]

File: django_simple_api/utils.py (hunk 3 above)
Comment (GJJ121785021) [reply django_simple_api/utils.py]: done

File: django_simple_api/utils.py [hunk 4]
```diff
@@ -49,3 +51,35 @@ def is_class_view(handler: Callable) -> bool:
...
+    urlpatterns: List[Union[URLPattern, URLResolver]],
+):
+    for item in urlpatterns:
+        if isinstance(item, URLPattern):
+            _wrapper_handler(wrappers, item.callback)
+        else:
+            wrapper_urlpatterns(wrappers, item.url_patterns)
+
+
+def wrapper_chain(wrappers: Sequence[Callable[[T], T]], view: Any) -> Any:
```
Comment (abersheeran) [inline django_simple_api/utils.py:73]: Maybe `wrapper_include` better than `wrapper_chain`.

Comment (GJJ121785021) [reply django_simple_api/utils.py]: done

**Review by abersheeran — DISMISSED** [review]

**Review by abersheeran — APPROVED** [review]

**Review by MeGustas-5427 — APPROVED** [review]

**Review by waitan2018 — APPROVED** [review]

**Pull request merged**